ml-agents/mlagents/trainers/components/reward_signals/gail/model.py

```diff
@@ -0,0 +1,261 @@
+from typing import Tuple, List
+
+import tensorflow as tf
+from mlagents.trainers.models import LearningModel
+
+
+class GAILModel(object):
+    def __init__(
+        self,
+        policy_model: LearningModel,
+        h_size: int = 128,
+        learning_rate: float = 3e-4,
+        encoding_size: int = 64,
+        use_actions: bool = False,
+    ):
+        """
+        The initializer for the GAIL reward generator.
+        https://arxiv.org/abs/1606.03476
```

Comment:
Do we want to reference VAIL as well? Since we are using it, it would probably help people reading out code to know why we are doing something so different from GAIL. Also, have we shown that VAIL does indeed outperform GAIL?

Reply:
Made VAIL an option. For simple stuff like Pyramids and Crawler, VAIL actually learns slower. I believe it makes learning more stable, but just from the learning speed on Pyramids and Crawler, I've decided to default to `use_vail = False`